beekeeper-metadata-cleanup/src/main/java/com/expediagroup/beekeeper/metadata/cleanup/handler/GenericMetadataHandler.java

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright (C) 2019-2020 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.beekeeper.metadata.cleanup.handler;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+
+import com.expediagroup.beekeeper.core.metadata.MetadataCleaner;
+import com.expediagroup.beekeeper.core.model.HousekeepingMetadata;
+import com.expediagroup.beekeeper.core.model.HousekeepingStatus;
+import com.expediagroup.beekeeper.core.model.LifecycleEventType;
+import com.expediagroup.beekeeper.core.path.PathCleaner;
+import com.expediagroup.beekeeper.core.repository.HousekeepingMetadataRepository;
+
+public abstract class GenericMetadataHandler {
+
+  private final Logger log = LoggerFactory.getLogger(GenericMetadataHandler.class);
+
+  public abstract HousekeepingMetadataRepository getHousekeepingMetadataRepository();
+
+  public abstract LifecycleEventType getLifecycleType();
+
+  public abstract MetadataCleaner getMetadataCleaner();
+
+  public abstract PathCleaner getPathCleaner();
+
+  public abstract Page<HousekeepingMetadata> findRecordsToClean(LocalDateTime instant, Pageable pageable);
+
+  public abstract Page<HousekeepingMetadata> findMatchingRecords(
+      String databaseName,
+      String tableName,
+      Pageable pageable);
+
+  /*-
```

Comment:
What does this body of comments apply to? One of the methods below or in general just to how this class behaves?

Reply:
just generally how the class behaves, I added it when starting the cleanup module as a helpful explanation for how the paging will work - its slightly different to the path cleanup. I can put it somewhere else or just take it out entirely if we think it's not needed.